### PR TITLE
3612_Handler Loop Shuffle

### PR DESCRIPTION
**Pull request by dinhquang251**
Handler button Loop,Shuffle
https://edu-redmine.framgia.vn/issues/3612

### CODE REVIEW

**Comment by dinhquang251** [comment]
ready

File: app/src/main/java/com/framgia/quangtran/music_42/mediaplayer/TracksPlayerManager.java [hunk 1]
```diff
@@ -73,54 +76,78 @@ public void initPlay(int position) {
...
         setState(StatePlayerType.STOP);
         mMediaPlayer.stop();
+        notifyStateChanged(getState());
     }
 
     @Override
     public void next() {
-        if (mTrackCurrentPosition < mTracks.size() - 1) {
-            mTrackCurrentPosition++;
+        if (mTrackPosition == mTracks.size() - 1 && getShuffleType() == ShuffleType.OFF) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/mediaplayer/TracksPlayerManager.java]: 1 is hard code. tương tự nơi khác

File: app/src/main/java/com/framgia/quangtran/music_42/service/TracksService.java [hunk 2]
```diff
@@ -167,15 +185,24 @@ public void onDestroy() {
...
+
+    @Override
+    public void notifyStateChanged(Track track, int stateType) {
+        switch (stateType) {
+            case ITracksPlayerManager.StatePlayerType.PAUSE:
+                updateDescriptionNotification(track);
+                break;
+            case ITracksPlayerManager.StatePlayerType.PLAYING:
+                updateDescriptionNotification(track);
+                break;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/service/TracksService.java:203]: add default case

File: app/src/main/java/com/framgia/quangtran/music_42/service/TracksService.java [hunk 3]
```diff
@@ -253,6 +286,35 @@ private void updateDescriptionNotification(Track track) {
...
+            case ACTION_PLAY:
+                changePlayPauseStatus();
+                break;
+            case ACTION_PAUSE:
+                changePlayPauseStatus();
+                break;
+            case ACTION_NEXT:
+                next();
+                break;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/service/TracksService.java:310]: add default case

**Comment by NguyenMinhTuanB** [comment]
commented

**Comment by dinhquang251** [comment]
ready

**Pull request merged**